crates/re_types_builder/src/objects.rs

```diff
@@ -0,0 +1,826 @@
+    /// Trim it yourself if needed!
+    ///
+    /// See also [`Docs::tagged_docs`].
+    pub doc: Vec<String>,
+
+    /// Tagged documentation for the object.
+    ///
+    /// Each entry maps a tag value to a bunch of lines.
+    /// Each entry in the vector is a raw line, extracted as-is from the fbs definition.
+    /// Trim it yourself if needed!
+    ///
+    /// E.g. the following will be associated with the `py` tag:
+    /// ```flatbuffers
+    /// /// \py Something something about how this fields behave in python.
+    /// my_field: uint32,
+    /// ```
+    ///
+    /// See also [`Docs::doc`].
+    pub tagged_docs: HashMap<String, Vec<String>>,
+}
+
+impl Docs {
+    fn from_raw_docs(
+        docs: Option<flatbuffers::Vector<'_, flatbuffers::ForwardsUOffset<&'_ str>>>,
+    ) -> Self {
+        // language-agnostic docs
+        let doc = docs
+            .into_iter()
+            .flat_map(|doc| doc.into_iter())
+            // NOTE: discard tagged lines!
+            .filter(|line| !line.trim().starts_with('\\'))
+            .map(ToOwned::to_owned)
+            .collect::<Vec<_>>();
+
+        // tagged docs, e.g. `\py this only applies to python!`
+        let tagged_docs = {
+            let tagged_lines = docs
+                .into_iter()
+                .flat_map(|doc| doc.into_iter())
+                // NOTE: discard _un_tagged lines!
+                .filter_map(|line| {
+                    let trimmed = line.trim();
+                    trimmed.starts_with('\\').then(|| {
+                        let tag = trimmed.split_whitespace().next().unwrap();
+                        let line = &trimmed[tag.len()..];
+                        (tag[1..].to_owned(), line.to_owned())
+                    })
+                })
+                .collect::<Vec<_>>();
+
+            let all_tags: HashSet<_> = tagged_lines.iter().map(|(tag, _)| tag).collect();
+            let mut tagged_docs = HashMap::new();
+
+            for cur_tag in all_tags {
+                tagged_docs.insert(
+                    cur_tag.clone(),
+                    tagged_lines
+                        .iter()
+                        .filter_map(|(tag, line)| (cur_tag == tag).then(|| line.clone()))
+                        .collect(),
+                );
+            }
+
+            tagged_docs
+        };
+
+        Self { doc, tagged_docs }
+    }
+}
+
+/// A high-level representation of a flatbuffers object, which can be either a struct, a union or
+/// an enum.
+#[derive(Debug, Clone)]
+pub struct Object {
+    /// File path of the associated fbs definition, e.g. `//rerun/components/point2d.fbs`.
+    pub filepath: String,
+
+    /// Fully-qualified name of the object, e.g. `rerun.components.Point2D`.
+    pub fqname: String,
+
+    /// Fully-qualified package name of the object, e.g. `rerun.components`.
+    pub pkg_name: String,
+
+    /// Name of the object, e.g. `Point2D`.
+    pub name: String,
```

Comment:
Ehhh I don't necessarily agree; it's pretty common to denormalize all of this.

But more importantly, if we went that route we would have to do so in a freestanding helper because most of the time we need `pkg_name` and `name` before we even have an `Object`, and then we'd need an actual method that forwards to the freestanding helper so that it's not impossible to find... that's a lot of hoops to jump through for something that's computed once during init and should never cause any trouble whatsoever.